interview-questions.md

```diff
@@ -8,42 +8,74 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+It seems that the biggest difference is that with OOP data and behavior are combined into objects while with with functional programing data and behavior are treated seperatley. **I also highly recommend the source that provided this answer it was really helpful)
 
 1. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: an interger is a whole number and float is a fraction of a number.
 
 Researched answer:
+(the source: https://www.digitalocean.com/community/tutorials/understanding-data-types-in-ruby)
+
+(source language: "Like in math, integers in computer programming are whole numbers"
+
+"A floating-point number or a float represents a real number. Real numbers can be either a rational or an irrational number; numbers that contain a fractional part")
+
+(My interpretation: an interger is a whole number that can be positive, negative or 0. When using large intergers that would normally use a comma insted we would use an underscore ie: 1,200,300 would read 1_200_300
+
+Floats are fractions of numbers that contain a decimal points that can also be positive or negative but cannot be 0 )
 
 1. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: an implicit return means that there is a built in return. This means that you do not need a return statement
 
 Researched answer:
+(the source: https://dev.to/ting682/implicit-vs-explicit-returns-14j5)
+
+(source language: "In Ruby, every method provides a return value. Ruby will return the last evaluated statement.")
+
+(My interpretation: Implicit return will automatically return the last line of every statement. this allows for less code and keeps the code cleaner)
 
 1. What is a block in Ruby?
 
-Your answer:
+Your answer: Blocks are anonymous functions that can be passed into methods
 
 Researched answer:
+(the source: https://www.tutorialspoint.com/ruby/ruby_blocks.htm#:~:text=Similarly%2C%20Ruby%20has%20a%20concept,as%20that%20of%20the%20block.)
+
+(source language: "A block looks similar to a method in Ruby. Methods consist of a name, but with blocks we don’t need to write a name, and always pass to a method call.
+
+Blocks are anonymous pieces of code that accept input from arguments and return a value.")
+
+(My interpretation: Blocks are a pieces of code that take in input from arguments and return a value. Blocks are usually defined by "do" and "end" but can also be defined using {} but only for a single line. i.e: my_number.times{'polyglots'} from our lesson on Ruby blocks and iterables.)
 
 1. How do you extract a value in a Ruby hash?
 
-Your answer:
+Your answer: You extract a value from a ruby hash by using bracket notation.
 
-Researched answer:
+Researched answer: I could not find a good source to explain this. Many of the sources demostrated how to extract a value but didn't really talk about using bracket notation. The only reference I could find was the lesson we did about hashes.
 
```

Comment:
I like that you included you research sources. I would suggest elaborating some more in your answers, you could throw in examples to show your full understanding more to the interviewer!